lk2nd/device/keys.c

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2022 Nikita Travkin <nikita@trvn.ru> */
+
+#include <debug.h>
+#include <libfdt.h>
+#include <stdlib.h>
+#include <string.h>
+#include <target.h>
+#include <sys/types.h>
+#include <dev/keys.h>
+
+#include <lk2nd/gpio.h>
+#include <lk2nd/keys.h>
+
+#include "device.h"
+
+#define LK2ND_MAX_KEYS 4
+
+struct lk2nd_keymap {
+	uint32_t keycode;
+	struct gpio_desc gpio;
+};
+
+static struct lk2nd_keymap keymap[LK2ND_MAX_KEYS] = {0};
+
+#if WITH_DEV_PMIC_PM8X41 || WITH_DEV_PMIC_PM8921
+extern uint32_t pm8x41_get_pwrkey_is_pressed(void);
+static int target_powerkey() { return pm8x41_get_pwrkey_is_pressed(); }
+#else
+static int target_powerkey() { return 0; }
+#endif
+
+/**
+ * lk2nd_keys_pressed() - Check if the key is pressed.
+ * @keycode: Keycode to check.
+ *
+ * The method will first check the override keymap and try to find a
+ * gpio attached to the keycode. Multiple physical keys can share the
+ * same keycode. Every key will be checked to find a pressed one and
+ * presence of only one override will cause the default key handlers
+ * to be disabled. If there is no override key for the code, the
+ * default target-specific key checks will be used for certain keys.
+ *
+ * Returns: 1 if the key is pressed, 0 if not.
+ *
+ * If the key is absent, 0 will be returned.
+ */
+int lk2nd_keys_pressed(uint32_t keycode)
```

Comment:
```suggestion
bool lk2nd_keys_pressed(uint32_t keycode)
```

also maybe `lk2nd_key_is_pressed`? but that's nitpick level, feel free to ignore

Reply:
Yeah it looks weird, but lk2nd_keys is a namespace for pressed() so unsure if we want inconsistency here